#!/usr/bin/env python
PACKAGE = "pico_zense_camera"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("depth_confidence_threshold", int_t, 0,
        "Depth distance confidance (0% keeps all, 100% drops every depth point with <100% confidence", 100,  0, 100)
gen.add("depth_range", double_t, 0, "Optimise the camera for this depth range", 1.0, 0, 5.6)

resolutions = gen.enum([gen.const("1920x1080", str_t, "1920x1080", "Max Resolution"),
                        gen.const("1280x720", str_t, "1280x720", "Default Resolution"),
                        gen.const("640x480", str_t, "640x480", "Depth Resolution"),
                        gen.const("640x360", str_t, "640x360", "Min Resolution")],
                     "List of supported resolutions")

gen.add("rgb_resolution", str_t, 0, "RGB Resolution", "1280x720", edit_method=resolutions)

exit(gen.generate(PACKAGE, "pico_zense_camera", "pico_zense_dcam710"))